mcs/class/System/System/MonoExeLocator.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace System {
+
+	internal static class MonoExeLocator {
+
+		public static string GacPath { get; private set; }
+		public static string MonoPath { get; private set; }
+		public static string McsPath { get; private set; }
+		public static string VbncPath { get; private set; }
+		public static string AlPath { get; private set; }
+
+		static MonoExeLocator () {
+
+			PropertyInfo gac = typeof (Environment).GetProperty ("GacPath", BindingFlags.Static | BindingFlags.NonPublic);
```

Comment:
This is not used in any assemblies that work on the mobile profiles. It's a refactoring of logic we already had scattered across the code base.